packages/ra-auth-cognito/src/Login.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+    Button,
+    CardContent,
+    CircularProgress,
+    Typography,
+} from '@mui/material';
+import * as React from 'react';
+import {
+    Form,
+    Login as RaLogin,
+    LoginFormClasses,
+    required,
+    TextInput,
+    useNotify,
+    useTranslate,
+} from 'react-admin';
+import clsx from 'clsx';
+import { SubmitHandler } from 'react-hook-form';
+import { LoginFormData, useCognitoLogin } from './useCognitoLogin';
+import { validatePasswordsMatch } from './validatePasswordsMatch';
+
+export const Login = (props: any) => {
+    return (
+        <RaLogin {...props}>
+            <LoginForm />
+        </RaLogin>
+    );
+};
+
+type NewPasswordFormData = {
+    newPassword: string;
+    confirmNewPassword: string;
+};
+
+type FormData = LoginFormData | NewPasswordFormData;
+
+export const LoginForm = (props: any) => {
+    const { redirectTo, className } = props;
+    const translate = useTranslate();
+    const notify = useNotify();
+    const [login, { isLoading, requireNewPassword }] = useCognitoLogin({
+        redirectTo,
+    });
+
+    const submit: SubmitHandler<FormData> = values => {
+        let finalValues = (values as NewPasswordFormData).confirmNewPassword
+            ? {
+                  newPassword: (values as NewPasswordFormData).newPassword,
+              }
+            : values;
```

Comment:
I have an error here.
After filling the `newPassword` and `confirmNewPassword`, `finalValues` gets set to only:

```js
              {
                  newPassword: (values as NewPasswordFormData).newPassword,
              }
```

And later `login(finalValues)` throw the following error:

```
Error: Username and Pool information are required.
    at new CognitoUser2 (CognitoUser.js:72:13)
    at authProvider.ts:107:24
    at new Promise (<anonymous>)
    at Object.login (authProvider.ts:91:20)
    at useLogin.ts:47:26
    at Object.mutationFn (useCognitoLogin.ts:69:17)
    at Object.fn (mutation.js:132:31)
    at run2 (retryer.js:95:31)
    at new Retryer2 (retryer.js:156:3)
    at Mutation2.executeMutation (mutation.js:126:20)
```

(because `username` is missing)